ls1intum/Artemis
# Feature/orion/show test output

### Checklist

### Motivation and Context


We wanted to display the test output (and build output in case of compile errors) in the Orion plugin in IntelliJ. This is the PR containing the Angular client changes necessary to forward this information to the IDE.

### Description

There is a new IDE build and test service, which gets called once a user submits code inside the IDE. This service then listens to any build results and forwards them accordingly to IntelliJ.
 
I also filter out any non programming exercises from the overview if the user opens the webpage using IntelliJ. **I do want to change this in a future PR**. I want to move this to a separate component, so that we don't have that many `*ngIf`s and have a dediated programming exercise overview just for IntelliJ (also including minor UI improvements like a smaller header since this takes too much space in the small IDE tool window) 

### Steps for Testing

**Use the current release of Orion: https://github.com/ls1intum/Orion/releases/tag/v0.2.3**

1. Open IntelliJ
2. Navigate to the setting as displayed in the gif and set the URL to the test server
3. Open/import any programming exercise
4. Click on submit
5. The test results should be displayed in a tool window after the build finished

### Screenshots



#### Where to find the settings
![orion_settings_new](https://user-images.githubusercontent.com/27979674/67581911-1b0cae80-f749-11e9-8989-8e6e1e2b6018.gif)


#### How to test
![orion_buildoutput](https://user-images.githubusercontent.com/27979674/66350418-868a0a00-e95b-11e9-986f-08df4b3a2ef4.gif)
